metron-platform/metron-writer/src/main/java/org/apache/metron/writer/BulkWriterResponseHandler.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.writer;
+
+import org.apache.metron.common.writer.BulkWriterResponse;
+
+/**
+ * This interface is used by the {@link org.apache.metron.writer.BulkWriterComponent} to report that a queue for a
+ * sensor type has been flushed.  Different frameworks may have different requirements for committing processed messages
+ * so this interface provides a way to pass in the appropriate commit logic for the framework in use.
+ */
+public interface BulkWriterResponseHandler {
+
+  /**
+   * Called immediately after a batch has been flushed.
+   * @param sensorType The type of sensor generating the messages
+   * @param response A response containing successes and failures within the batch
+   */
+  void handleFlush(String sensorType, BulkWriterResponse response);
```

Comment:
I think they are different because they are called at different times and have different purposes:  `'FlushPolicy` classes are called before a flush to determine if a flush is necessary and `BulkWriterResponseHandler` is called after a flush happens with the results.  Maybe an example can help me understand?

Reply:
It may just be a bad idea. :)  They just seem so similar to me.  I'll flesh out something and see if it makes sense.  Don't worry about it for now.

Reply:
Just to flush out the idea, i opened a PR against your branch.  This is only to flesh out the idea, I didn't attempt to get the tests all working.  I added some comments in there to help emphasize the changes, since a bunch of master commits showed up in the PR.

https://github.com/merrimanr/incubator-metron/pull/40

Feel free to take it or leave it.  It is a subtle difference that in my mind makes things simpler to test and reason about.  But I would not be against keeping what you have.  It is completely your call.

Reply:
The latest commit includes your suggestion.  Let me know what you think.